hie-plugin-api/Haskell/Ide/Engine/Ghc.hs

```diff
@@ -130,131 +129,188 @@ srcErrToDiag df rfm se = do
   (diags, errs) <- processMsgs errMsgs
   return (Diagnostics diags, errs)
 
--- ---------------------------------------------------------------------
 
-myWrapper :: GM.IOish m
+-- | Run a Ghc action and capture any diagnostics and errors produced.
+captureDiagnostics :: (MonadIO m, GhcMonad m)
   => (FilePath -> FilePath)
-  -> GM.GmlT m ()
-  -> GM.GmlT m (Diagnostics, AdditionalErrs)
-myWrapper rfm action = do
+  -> m r
+  -> m (Diagnostics, AdditionalErrs, Maybe r)
+captureDiagnostics rfm action = do
   env <- getSession
-  diagRef <- liftIO $ newIORef mempty
+  diagRef <- liftIO $ newIORef $ Diagnostics mempty
   errRef <- liftIO $ newIORef []
   let setLogger df = df { log_action = logDiag rfm errRef diagRef }
-      setDeferTypedHoles = setGeneralFlag' Opt_DeferTypedHoles
+      unsetWErr df = unSetGeneralFlag' Opt_WarnIsError (emptyFatalWarningFlags df)
+      unsetMissingHomeModules = flip wopt_unset Opt_WarnMissingHomeModules
+      setRawTokenStream = setGeneralFlag' Opt_KeepRawTokenStream
```

Comment:
Unclear whether these options should go here or in `hie-bios`.

Reply:
I think here is fine, anybody should be able to set their preferred set of dynflags?